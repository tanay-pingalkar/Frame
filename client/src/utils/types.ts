export interface tokenData {
  register?: subTokenData;
  login?: subTokenData;
  googleLogin?: subTokenData;
}
export interface authData {
  auth?: {
    ErrorMsg: string | null;
    user: user;
  };
}
export interface user {
  id?: number;
  name?: string;
  email?: string;
}
export interface subTokenData {
  ErrorMsg: string | null;
  token: string | null;
}

export interface action {
  type: string;
  payload?: any;
}

export interface userAction extends action {
  payload?: user;
}

export type where = "home" | "notification" | "search" | "profile" | "upload";
type Omit<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>>;
export interface whereActtion extends Omit<action, "payload"> {
  type: where;
}

export interface frame {
  likeNumber: number;
  isLiked: boolean;
  frame: {
    title: string;
    frame: string;
    description: string;
    user: user;
    id: string;
  };
}

export type frames = Array<frame>;

export interface getFramesType {
  getFrames: frames;
}

export type isAuth = "nolog" | "logging" | "logged";
export interface state {
  isAuth: isAuth;
  userInfo: {
    id: string;
    name: string;
    email: string;
  };
}
export type setStateString = (value: React.SetStateAction<string>) => void;
export type setStateJsx = (value: React.SetStateAction<boolean>) => void;
