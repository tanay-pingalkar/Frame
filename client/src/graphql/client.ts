import { GraphQLClient } from "graphql-request";

export const client = new GraphQLClient("http://localhost:4000/graphql");
