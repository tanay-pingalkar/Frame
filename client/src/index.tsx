import ReactDOM from "react-dom";
import "./styles/index.scss";
import App from "./App";
import { createStore } from "redux";
import allReducer from "./redux/combine";
import { Provider } from "react-redux";
import { composeWithDevTools } from "redux-devtools-extension";

const store = createStore(allReducer, composeWithDevTools());

ReactDOM.render(
  <Provider store={store}>
    <App />
  </Provider>,
  document.getElementById("root")
);
