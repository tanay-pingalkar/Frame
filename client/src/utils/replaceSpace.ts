import { React } from "@ungap/global-this";

export const replace = (
  e: React.ChangeEvent<HTMLInputElement>,
  setName: React.Dispatch<React.SetStateAction<string>>
): void => {
  let val: string = e.target.value;
  val = val.replace(/\s/g, "-");
  setName(val);
};
