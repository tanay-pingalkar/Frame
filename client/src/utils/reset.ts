export const reset = () => {
  const cards = document.querySelectorAll<HTMLElement>(".card");
  cards.forEach((child) => {
    child.className = "card";
    child.style.animation = "";
  });
};
