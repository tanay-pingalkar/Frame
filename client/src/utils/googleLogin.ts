import { GoogleLoginResponse } from "react-google-login";
import { client } from "../graphql/client";
import { GOOGLE_AUTH } from "../graphql/mutations/googleAuth";
import { setStateString, tokenData } from "./types";

export const handleGoogle = async (
  googleData: GoogleLoginResponse,
  seterror: setStateString
): Promise<boolean> => {
  // store returned user somehow
  let token: tokenData = {};
  try {
    token = await client.request<tokenData>(GOOGLE_AUTH, {
      token: googleData.tokenId,
    });
    if (token.googleLogin?.ErrorMsg) {
      seterror("*" + token.googleLogin?.ErrorMsg);
      return false;
    } else if (token.googleLogin?.token) {
      localStorage.setItem("TOKEN", token.googleLogin?.token);
      return true;
    } else {
      return false;
    }
  } catch (error) {
    console.log(error);
    return false;
  }
};
