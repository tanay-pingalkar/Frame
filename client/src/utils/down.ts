import { frames } from "../utils/types";
import { reset } from "./reset";

export const useDown = (
  frames: frames,
  seti: React.Dispatch<React.SetStateAction<number>>,
  i: number
): (() => boolean) => {
  return (): boolean => {
    if (frames.length !== 0 && i > 2) {
      reset();
      // selecting elements
      const upcomingAbove = document.querySelectorAll(".card")[
        i - 3
      ] as HTMLElement;
      const above = document.querySelectorAll(".card")[i - 2] as HTMLElement;
      const view = document.querySelectorAll(".card")[i - 1] as HTMLElement;
      const below = document.querySelectorAll(".card")[i] as HTMLElement;

      //adding class to elements
      above?.classList.add("above");
      view?.classList.add("view");
      below?.classList.add("below");
      upcomingAbove?.classList.add("upcomingAbove");

      //ading animation
      if (above) above.style.animation = "aboveDown 0.5s ease forwards";
      if (view) view.style.animation = "viewDown 0.5s ease forwards";
      if (below) below.style.animation = "belowDown 0.5s ease forwards";
      seti(i - 1);
      return true;
    }
    return false;
  };
};
