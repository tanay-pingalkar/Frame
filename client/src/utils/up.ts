import { frames } from "../utils/types";
import { reset } from "./reset";

export const up = (i: number): void => {
  reset();
  const above = document.querySelectorAll(".card")[i] as HTMLElement;
  const view = document.querySelectorAll(".card")[i + 1] as HTMLElement;
  const below = document.querySelectorAll(".card")[i + 2] as HTMLElement;

  //adding class to elements
  above?.classList.add("above");
  view?.classList.add("view");
  below?.classList.add("below");

  //ading animation
  if (above) above.style.animation = "aboveUp 0.5s ease forwards";
  if (view) view.style.animation = "viewUp 0.5s ease forwards";
  if (below) below.style.animation = "belowUp 0.5s ease forwards";
};

export const useUp = (
  frames: frames,
  seti: React.Dispatch<React.SetStateAction<number>>,
  i: number,
  setNoWayBack: React.Dispatch<React.SetStateAction<number>>,
  noWayBack: number,
  setlastframe: React.Dispatch<React.SetStateAction<number>>
): (() => boolean) => {
  return (): boolean => {
    if (frames.length !== 0 && frames.length - 2 > i) {
      up(i);
      seti(i + 1);
      if (i === noWayBack) setNoWayBack(noWayBack + 1);
      if (noWayBack % 4 === 0)
        setlastframe(Number(frames[frames.length - 1].frame.id));
      return true;
    }
    return false;
  };
};
