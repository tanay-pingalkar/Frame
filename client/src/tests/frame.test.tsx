import Frame from "../components/frame";
import React from "react";
import { Provider } from "react-redux";
import configureMockStore, { MockStoreEnhanced } from "redux-mock-store";
import thunk from "redux-thunk";
import {
  cleanup,
  render,
  screen,
  fireEvent,
  act,
  waitFor,
} from "@testing-library/react";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import { client } from "../graphql/client";
import { frame } from "../utils/types";

const mockStore = configureMockStore([thunk]);

jest.mock("../graphql/client");

describe("testing frame component", () => {
  let store: MockStoreEnhanced<unknown, {}>;
  let data: frame;
  beforeEach(() => {
    store = mockStore({
      userInfo: {
        id: "1",
      },
    });
    data = {
      isLiked: false,
      likeNumber: 100,

      frame: {
        title: "title",
        description: "description",
        frame: "frame",
        user: {
          name: "user",
        },
        id: "5",
      },
    };
    render(
      <Provider store={store}>
        <Frame frame={data} />
      </Provider>
    );
  });

  afterEach(() => {
    jest.clearAllMocks();
    cleanup();
  });

  it("should render frame properly", () => {
    const item = screen.getByTestId("card");
    expect(item).toBeInTheDocument();
    expect(item.className).toBe("card");
  });

  it("should render title properly", () => {
    const item = screen.getByTestId("title");
    expect(item).toBeInTheDocument();
    expect(item).toHaveTextContent("title");
    expect(item.className).toBe("title");
  });
  it("should render name properly", () => {
    const item = screen.getByTestId("name");
    expect(item).toBeInTheDocument();
    expect(item).toHaveTextContent("user");
    expect(item.className).toBe("name");
  });
  it("should render image properly", () => {
    const item = screen.getByTestId("img");
    expect(item).toBeInTheDocument();
    expect(item.className).toBe("img");
  });

  it("should render like properly", async () => {
    const main = screen.getByTestId("like");
    const item = screen.getByTestId("like-num");
    expect(item).toBeInTheDocument();
    expect(item).toHaveTextContent("100".toString());
    const mockRes = { like: { like: true, errorMsg: null } };
    // @ts-ignore
    client.request.mockResolvedValue(mockRes);
    act(() => {
      fireEvent.click(main);
    });

    await waitFor(() => {
      expect(main).toHaveTextContent("101");
    });

    const mocknewRes = { like: { like: false, errorMsg: "disliked" } };

    // @ts-ignore
    client.request.mockResolvedValue(mocknewRes);
    act(() => {
      fireEvent.click(main);
    });

    await waitFor(() => {
      expect(main).toHaveTextContent("100");
    });
  });

  it("should match snapshot", () => {
    const frame = renderer
      .create(
        <Provider store={store}>
          <Frame frame={data} />
        </Provider>
      )
      .toJSON();
    expect(frame).toMatchSnapshot();
  });

  it("should match snapshot when true", () => {
    const trueData = data;
    trueData.isLiked = true;
    const frame = renderer
      .create(
        <Provider store={store}>
          <Frame frame={trueData} />
        </Provider>
      )
      .toJSON();
    expect(frame).toMatchSnapshot();
  });
});
