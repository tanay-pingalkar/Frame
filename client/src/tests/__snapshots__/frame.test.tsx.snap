// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing frame component should match snapshot 1`] = `
<div
  className="card"
  data-testid="card"
>
  <p
    className="name"
    data-testid="name"
  >
    user
  </p>
  <p
    className="title"
    data-testid="title"
  >
    title
  </p>
  <img
    alt="lol"
    className="img"
    data-testid="img"
    src="frame"
  />
  <p
    className="description"
    data-testid="description"
  >
    description
  </p>
  <span
    data-testid="like"
    onClick={[Function]}
  >
    <svg
      fill="none"
      height="20"
      viewBox="0 0 20 20"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9.99998 18.8873C9.71525 18.8873 9.44075 18.7842 9.22682 18.5968C8.41887 17.8903 7.63991 17.2264 6.95265 16.6408L6.94914 16.6377C4.93423 14.9207 3.19427 13.4378 1.98364 11.9771C0.630339 10.3441 0 8.79578 0 7.10434C0 5.46097 0.563506 3.94485 1.58661 2.83508C2.62191 1.71219 4.0425 1.09375 5.58715 1.09375C6.74163 1.09375 7.7989 1.45874 8.72954 2.1785C9.1992 2.54181 9.62492 2.98645 9.99998 3.5051C10.3752 2.98645 10.8008 2.54181 11.2706 2.1785C12.2012 1.45874 13.2585 1.09375 14.413 1.09375C15.9575 1.09375 17.3782 1.71219 18.4135 2.83508C19.4366 3.94485 20 5.46097 20 7.10434C20 8.79578 19.3698 10.3441 18.0165 11.9769C16.8058 13.4378 15.066 14.9205 13.0514 16.6374C12.363 17.224 11.5828 17.8889 10.773 18.5971C10.5592 18.7842 10.2846 18.8873 9.99998 18.8873ZM5.58715 2.26532C4.37362 2.26532 3.25881 2.74963 2.44781 3.62915C1.62475 4.52194 1.17141 5.75607 1.17141 7.10434C1.17141 8.52692 1.70013 9.79919 2.88558 11.2296C4.03136 12.6122 5.73562 14.0645 7.70888 15.7462L7.71254 15.7492C8.40239 16.3371 9.1844 17.0036 9.9983 17.7153C10.8171 17.0023 11.6003 16.3347 12.2915 15.7458C14.2647 14.0642 15.9688 12.6122 17.1145 11.2296C18.2998 9.79919 18.8285 8.52692 18.8285 7.10434C18.8285 5.75607 18.3752 4.52194 17.5522 3.62915C16.7413 2.74963 15.6263 2.26532 14.413 2.26532C13.524 2.26532 12.7078 2.54791 11.9871 3.10516C11.3449 3.60199 10.8975 4.23004 10.6352 4.66949C10.5003 4.89548 10.2629 5.03036 9.99998 5.03036C9.73707 5.03036 9.49965 4.89548 9.36476 4.66949C9.10261 4.23004 8.65523 3.60199 8.01283 3.10516C7.29216 2.54791 6.47597 2.26532 5.58715 2.26532Z"
        fill="black"
      />
    </svg>
    <p
      data-testid="like-num"
    >
      100
    </p>
  </span>
</div>
`;

exports[`testing frame component should match snapshot when true 1`] = `
<div
  className="card"
  data-testid="card"
>
  <p
    className="name"
    data-testid="name"
  >
    user
  </p>
  <p
    className="title"
    data-testid="title"
  >
    title
  </p>
  <img
    alt="lol"
    className="img"
    data-testid="img"
    src="frame"
  />
  <p
    className="description"
    data-testid="description"
  >
    description
  </p>
  <span
    data-testid="like"
    onClick={[Function]}
  >
    <svg
      fill="none"
      height="20"
      viewBox="0 0 20 20"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M14.6875 1.17188C13.6022 1.17188 12.6072 1.51578 11.7302 2.19406C10.8894 2.84434 10.3296 3.67258 10 4.27484C9.67043 3.67254 9.11063 2.84434 8.2698 2.19406C7.39277 1.51578 6.39777 1.17188 5.3125 1.17188C2.28391 1.17188 0 3.6491 0 6.93414C0 10.4831 2.84934 12.9113 7.16285 16.5872C7.89535 17.2114 8.72563 17.919 9.58859 18.6737C9.70234 18.7733 9.84844 18.8281 10 18.8281C10.1516 18.8281 10.2977 18.7733 10.4114 18.6737C11.2745 17.9189 12.1047 17.2114 12.8376 16.5868C17.1507 12.9113 20 10.4831 20 6.93414C20 3.6491 17.7161 1.17188 14.6875 1.17188Z"
        fill="black"
      />
    </svg>
    <p
      data-testid="like-num"
    >
      100
    </p>
  </span>
</div>
`;
